src/github_type.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Type of returned data from GitHub web API."""
+
+
+from typing import List, TypedDict
+
+
+class RunnerApplication(TypedDict, total=False):
+    """Information on a single runner application."""
+
+    os: str
+    architecture: str
+    download_url: str
+    filename: str
+    sha256_checksum: str  # NotRequired
```

Comment:
I am currently trying out the `NotRequired` from: https://peps.python.org/pep-0655/.

Imported it from `typing_extensions`, since the lib is already used elsewhere.